controlplane/kubeadm/internal/workload_cluster_etcd.go

```diff
@@ -202,7 +202,7 @@ func (w *Workload) ForwardEtcdLeadership(ctx context.Context, machine *clusterv1
 		return errors.Wrap(err, "failed to list control plane nodes")
 	}
 
-	etcdClient, err := w.etcdClientGenerator.forLeader(ctx, nodes)
+	etcdClient, err := w.etcdClientGenerator.forAny(ctx, nodes)
```

Comment:
Does this work with `forAny`? Or should this still go directly to the current leader?

Reply:
Forwarding leadership definitely requires the current leader. While the current implementation eventually works, it will still be cleaner to stick to `forLeader` in this call.